### PR TITLE
Feature/auth proxy

### PR DESCRIPTION
**Pull request by weilaaa**
Ⅰ. Describe what this PR does

make an auth proxy to access to k8s

Ⅱ. Does this pull request fix one issue?

#62

### CODE REVIEW

File: pkg/authentication/authenticators/jwt/jwt.go [hunk 1]
```diff
@@ -110,10 +110,16 @@ func (a *AuthJwt) Authentication(token string) (user *v1beta1.UserInfo, err erro
 	return nil, fmt.Errorf("invaild token")
 }
 
-func (a *AuthJwt) RefreshToken(token string) (string, error) {
+func (a *AuthJwt) RefreshToken(token string) (*v1beta1.UserInfo, string, error) {
 	userInfo, err := a.Authentication(token)
 	if err != nil {
-		return "", fmt.Errorf("parse token error: %s", err)
+		return nil, "", fmt.Errorf("parse token error: %s", err)
```
Comment (fusida) [inline pkg/authentication/authenticators/jwt/jwt.go:116]: err should be %v

File: pkg/warden/server/authproxy/authproxy.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// parse token transfer to user info
+	user, err := token.GetUserFromReq(r)
+	if err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
+		fmt.Fprintf(w, "token invalid: %v", err)
+		return
+	}
+
+	clog.Debug("user(%v) access to %v with verb(%v)", user.Username, r.Method)
```
Comment (fusida) [inline pkg/warden/server/authproxy/authproxy.go:97]: 3 %v to 2 params

**Review by fusida — APPROVED** [review]
LGTM

**Review by JiahuiZhao11 — COMMENTED** [review]
LGTM

**Review by astargeeker — APPROVED** [review]
LGTM

**Pull request merged**